include/ignition/rendering/base/BaseAxisVisual.hh

```diff
@@ -51,25 +63,52 @@ namespace ignition
     {
     }
 
+    //////////////////////////////////////////////////
+    template <class T>
+    void BaseAxisVisual<T>::SetLocalScale(double _x, double _y, double _z)
+    {
+      xArrow->SetLocalScale(_x, _y, _z);
+      yArrow->SetLocalScale(_x, _y, _z);
+      zArrow->SetLocalScale(_x, _y, _z);
+    }
+
+    //////////////////////////////////////////////////
+    template <class T>
+    void BaseAxisVisual<T>::SetLocalScale(double _scale)
+    {
+      xArrow->SetLocalScale(_scale, _scale, _scale);
+      yArrow->SetLocalScale(_scale, _scale, _scale);
+      zArrow->SetLocalScale(_scale, _scale, _scale);
+    }
+
+    //////////////////////////////////////////////////
+    template <class T>
+    void BaseAxisVisual<T>::SetLocalScale(const math::Vector3d &_scale)
+    {
+      xArrow->SetLocalScale(_scale.X(), _scale.Y(), _scale.Z());
```

Comment:
how about:

```
      for (unsigned int i = 0; i < this->ChildCount(); ++i)
        this->ChildByIndex(i)->SetLocalScale(_scale);
```

this way we don't need to declare the private `xArrow`, `yArrow`, `zArrow` variables